examples/step-87/step-87.cc

```diff
@@ -0,0 +1,1175 @@
+      vector);
+
+    // We create a list of points inside the domain at which we
+    // would like to evaluate the vector.
+    const auto points_line =
+      create_points_along_line(Point<dim>(0.0, 0.5), Point<dim>(1.0, 0.5), 20);
+
+    // Now, we loop over all evaluation points. In the first step, we determine
+    // via GridTools::find_active_cell_around_point() the cell $K$ that
+    // surrounds the point and translate the given real coordinate
+    // $\boldsymbol{x}$ to the corresponding coordinate on the unit cell
+    // $\hat{\boldsymbol{x}}_K$ according to the provided mapping.
+    // The resulting information is printed to the screen.
+    std::vector<double> values_line;
+    values_line.reserve(points_line.size());
+
+    for (const auto &p_real : points_line)
+      {
+        const auto [cell, p_unit] =
+          GridTools::find_active_cell_around_point(mapping,
+                                                   dof_handler,
+                                                   p_real);
+
+        {
+          AssertThrow(cell != dof_handler.end(), ExcInternalError());
+          std::cout << " - Found point with real coordinates: " << p_real
+                    << std::endl;
+          std::cout << "   - in cell with vertices:";
+          for (const auto &v : cell->vertex_indices())
+            std::cout << " (" << cell->vertex(v) << ")";
+          std::cout << std::endl;
+          std::cout << "   - with coordinates on the unit cell: (" << p_unit
+                    << ")" << std::endl;
+        }
+
+        // Now that we have determined $K$ and $\hat{\boldsymbol{x}}_K$, we can
+        // perform the evaluation of the finite element solution at this
+        // point. In the following, we show three approaches for this
+        // purpose. In the first approach, we follow a traditional technique
+        // by using FEValues based on a cell-specific quadrature rule
+        // consisting of the unit point.
+        std::cout << " - Values at point:" << std::endl;
+
+        {
+          FEValues<dim> fe_values(mapping,
+                                  fe,
+                                  Quadrature<dim>(p_unit),
+                                  update_values);
+          fe_values.reinit(cell);
+
+          std::vector<double> quad_values(1);
+          fe_values.get_function_values(vector, quad_values);
+          const double value_0 = quad_values[0];
+          std::cout << "  - " << value_0 << " (w. FEValues)" << std::endl;
+          values_line.push_back(value_0);
+        }
+
+        // The second approach considers FEPointEvaluation, which directly
+        // takes a list of unit points for the subsequent evaluation.
+        {
+          std::vector<double> cell_vector(fe.n_dofs_per_cell());
+          cell->get_dof_values(vector, cell_vector.begin(), cell_vector.end());
+
+          FEPointEvaluation<1, dim> fe_point(mapping, fe, update_values);
```

Comment:
I guess you do this for demonstration purposes, but in general the recommendation would be to move `FEPointEvaluation` and `cell_vector` out of the cell loop, because they contain relatively expensive intialization work that does not need to be repeated.